### ISSUE TITLE
Error getting result from polling source java.sql.SQLSyntaxErrorException: Table 'cattle.setting' doesn't exist

### ISSUE THREAD

**Issue by gghonor** [rancher/rancher#391]
Sometimes I got this ERROR, too.
What's is problem?

2015-04-01 08:52:37,944 INFO    [main] [ConsoleStatus] [1/104] [0ms] [0ms] Loading bootstrap 
2015-04-01 08:52:38,263 INFO    [main] [ConsoleStatus] [2/104] [319ms] [298ms] Loading config-defaults 
2015-04-01 08:53:06,829 ERROR [:] [] [] [] [igurationSource] [c.n.config.AbstractPollingScheduler ] Error getting result from polling source java.sql.SQLSyntaxErrorException: Table 'cattle.setting' doesn't exist
	at org.mariadb.jdbc.internal.SQLExceptionMapper.get(SQLExceptionMapper.java:138) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.internal.SQLExceptionMapper.throwException(SQLExceptionMapper.java:106) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.executeQueryEpilog(MySQLStatement.java:264) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.execute(MySQLStatement.java:288) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.executeQuery(MySQLStatement.java:302) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLPreparedStatement.executeQuery(MySQLPreparedStatement.java:112) ~[mariadb-java-client-1.1.7.jar:na]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:96) ~[commons-dbcp-1.4.jar:1.4]
	at org.apache.commons.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:96) ~[commons-dbcp-1.4.jar:1.4]
	at com.netflix.config.sources.JDBCConfigurationSource.load(JDBCConfigurationSource.java:107) ~[archaius-core-0.5.14.jar:na]
	at com.netflix.config.sources.JDBCConfigurationSource.poll(JDBCConfigurationSource.java:88) ~[archaius-core-0.5.14.jar:na]
	at io.cattle.platform.archaius.sources.LazyJDBCSource.poll(LazyJDBCSource.java:30) ~[cattle-framework-archaius-0.5.0-SNAPSHOT.jar:na]
	at com.netflix.config.AbstractPollingScheduler$1.run(AbstractPollingScheduler.java:163) ~[archaius-core-0.5.14.jar:na]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:471) [na:1.7.0_75]
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304) [na:1.7.0_75]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178) [na:1.7.0_75]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [na:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_75]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_75]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_75]
Caused by: org.mariadb.jdbc.internal.common.QueryException: Table 'cattle.setting' doesn't exist
	at org.mariadb.jdbc.internal.mysql.MySQLProtocol.getResult(MySQLProtocol.java:942) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.internal.mysql.MySQLProtocol.executeQuery(MySQLProtocol.java:991) ~[mariadb-java-client-1.1.7.jar:na]
	at org.mariadb.jdbc.MySQLStatement.execute(MySQLStatement.java:281) ~[mariadb-java-client-1.1.7.jar:na]
	... 15 common frames omitted

**Comment by corn13read** [comment]
Same issue here:
```
2015-04-14 22:16:31,778 ERROR [:] [] [] [] [igurationSource] [c.n.config.AbstractPollingScheduler ] Error getting re
sult from polling source java.sql.SQLSyntaxErrorException: Table 'cattle.setting' doesn't exist
        at org.mariadb.jdbc.internal.SQLExceptionMapper.get(SQLExceptionMapper.java:138) ~[mariadb-java-client-1.1.7
.jar:na]
        at org.mariadb.jdbc.internal.SQLExceptionMapper.throwException(SQLExceptionMapper.java:106) ~[mariadb-java-c
lient-1.1.7.jar:na]
        at org.mariadb.jdbc.MySQLStatement.executeQueryEpilog(MySQLStatement.java:264) ~[mariadb-java-client-1.1.7.j
ar:na]
        at org.mariadb.jdbc.MySQLStatement.execute(MySQLStatement.java:288) ~[mariadb-java-client-1.1.7.jar:na]
        at org.mariadb.jdbc.MySQLStatement.executeQuery(MySQLStatement.java:302) ~[mariadb-java-client-1.1.7.jar:na]
        at org.mariadb.jdbc.MySQLPreparedStatement.executeQuery(MySQLPreparedStatement.java:112) ~[mariadb-java-clie
nt-1.1.7.jar:na]
        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:96) ~[c
ommons-dbcp-1.4.jar:1.4]
        at org.apache.commons.dbcp.DelegatingPreparedStatement.executeQuery(DelegatingPreparedStatement.java:96) ~[c
ommons-dbcp-1.4.jar:1.4]
        at com.netflix.config.sources.JDBCConfigurationSource.load(JDBCConfigurationSource.java:107) ~[archaius-core
-0.5.14.jar:na]
        at com.netflix.config.sources.JDBCConfigurationSource.poll(JDBCConfigurationSource.java:88) ~[archaius-core
```

**Comment by ibuildthecloud** [comment]
@gghonor @corn13read Do you consistently get that error?  What OS and docker version are you running?

**Comment by vincent99** [comment]
@deniseschannon also got this in #581

**Comment by yunspace** [comment]
I also get this error using:
- Rancer.io v0.16.3 on Azure Ubuntu VM
- Created a empty `cattle` mysql database running in another container. 

My steps are: 
1. I started the rancher.io without starting up MySQL container. It polled continuously for the mysql but can't obtain a connection
2. I started up the MySQL container.
3. Rancho can connect but start to throw the `cattle.setting doesn't exist` error.
4. Rancho starts up, and I can verify the database was seeded correctly.
5. I restart Rancho container, but don't get the error anymore. Also all my data have persisted correctly in mysql

**Comment by ibuildthecloud** [comment]
Between steps 1-4 do you ever restart rancher/server.  Is the only restart in step 5?

**Comment by yunspace** [comment]
@ibuildthecloud there were no restarts between 1-4. The whole exercise was trying to test if the DB persistence was working. After step 4 I added github authentication and registered all my hosts, then restarted.

**Comment by EtreRe** [comment]
Same issue for me on Ubuntu 14.04 64 bits LTS (HVM) on Gandi Hosting
I can give you an access to the VM if you want to look ?

```
# docker -v
Docker version 1.6.0, build 4749651
```

```
# uname -a
Linux server01 3.12.36-guest-27-c9b6624-x86_64 #2 SMP Fri Jan 16 15:04:50 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by amithasson** [comment]
Same here.

docker version
############
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 4749651
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 4749651
OS/Arch (server): linux/amd64

docker info
#########
Containers: 1
Images: 26
Storage Driver: aufs
 Root Dir: /var/lib/docker/aufs
 Backing Filesystem: extfs
 Dirs: 28
 Dirperm1 Supported: false
Execution Driver: native-0.2
Kernel Version: 3.13.0-24-generic
Operating System: Ubuntu 14.04.1 LTS
CPUs: 2
Total Memory: 7.798 GiB
Name: myd-vm00483
ID: DFWH:S4WA:IKRW:P7SM:ZEB4:HQ2C:ONVF:AOID:OG5R:WANE:TDJT:OSC5
Http Proxy: http://hahsed-url.hp.com:8080
Https Proxy: http://hahsed-url.hp.com:8080
WARNING: No swap limit support

uname -a:
########
Linux myd-vm00483 3.13.0-24-generic #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

**Comment by will-chan** [comment]
obsolete.

**Comment by andrewmclagan** [comment]
obsolete?

`v1.1.0-dev4`

Receiving this error on pulling the latest tag and trying to start rancher container with external DB:

````
sudo docker run -d --restart=always -p 80:80 \
    -e CATTLE_DB_CATTLE_MYSQL_HOST=XXXX \
    -e CATTLE_DB_CATTLE_MYSQL_PORT=3306 \
    -e CATTLE_DB_CATTLE_MYSQL_NAME=rancher \
    -e CATTLE_DB_CATTLE_USERNAME=root \
    -e CATTLE_DB_CATTLE_PASSWORD=XXXX \
    rancher/server
````

**Comment by deniseschannon** [comment]
@andrewmclagan Is that the exact command you used? You would need `-p 80:8080` cause we have exposed port `8080` in our container.

**Comment by andrewmclagan** [comment]
Yep using 80:8080

I'm using external MariaDB. Will that be the issue?

**Comment by deniseschannon** [comment]
MariaDB is MySQL based so it should work.

**Comment by MetalArend** [comment]
Still getting this running the rancher test fig yml file, mentioned here: https://github.com/rancher/rancher/blob/master/tests/server/fig-test-env.yml

I only added a data container for data persistence to the mysqlmanual service, and it keeps throwing the error mentioned in the original issue. Is there a new issue to follow, or is this one closed too early?

**Comment by deniseschannon** [comment]
@MetalArend Can you open a new issue with details of what you did adn the logs that you are getting from it?

**Comment by gcristofol** [comment]
Same thing here. It failed for me bootstrapping 1.6.2 but I even went back to 1.5.2 and had similar results
```
2017-06-23 02:35:06,610 ERROR [:] [] [] [] [igurationSource] [c.n.config.AbstractPollingScheduler ] Error getting result from polling source java.sql.SQLSyntaxErrorException: (conn:65525) Table 'cattle.setting' doesn't exist
Query is: SELECT distinct name, value FROM setting
```

**Comment by blaggacao** [comment]
Here we go, full logs from start to fail: https://gist.github.com/blaggacao/cda58c151168ad8b41c95c4415483eee
I'm working with Google Cloud SQL

Note line:
`2017-07-15 22:02:17,704 ERROR [:] [] [] [] [igurationSource] [c.n.config.AbstractPollingScheduler ] Error getting result from polling source java.sql.SQL
SyntaxErrorException: (conn:2107) Table 'cattle.setting' doesn't exist`

**Comment by blaggacao** [comment]
I'm searching in the dark, but "db lock" sounds awful.
```sql
mysql> SELECT * FROM DATABASECHANGELOGLOCK                         ;
+----+--------+---------------------+---------------------------+
| ID | LOCKED | LOCKGRANTED         | LOCKEDBY                  |
+----+--------+---------------------+---------------------------+
|  1 |       | 2017-07-15 22:18:43 | 0a11622d11db (172.18.0.2) |
+----+--------+---------------------+---------------------------+
1 row in set (0.00 sec)
```

**Comment by superseb** [comment]
Sounds similar as https://github.com/rancher/rancher/issues/8962, can you continue there and does it complete eventually?

**Comment by blaggacao** [comment]
@superseb Thanks, indeed I'll shift there for follow up.
